Antalya 25.6 - CICD / version related fixes

Enmk opened the pull request:
Port of #854 

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)


#### Exclude tests:
- [ ] <!---ci_exclude_fast--> Fast test
- [x] <!---ci_exclude_integration--> Integration Tests
- [ ] <!---ci_exclude_stateless--> Stateless tests
- [ ] <!---ci_exclude_stateful--> Stateful tests
- [x] <!---ci_exclude_performance--> Performance tests
- [x] <!---ci_exclude_asan--> All with ASAN
- [x] <!---ci_exclude_tsan--> All with TSAN
- [x] <!---ci_exclude_msan--> All with MSAN
- [x] <!---ci_exclude_ubsan--> All with UBSAN
- [x] <!---ci_exclude_coverage--> All with Coverage
- [ ] <!---ci_exclude_aarch64|arm--> All with Aarch64
- [x] <!---ci_exclude_regression--> All Regression
- [x] <!---no_ci_cache--> Disable CI Cache

Enmk committed:
Merge pull request #844 from Altinity/version_debug

25.3 Antalya - Some debugging regarding generated CH version

Enmk committed:
Merge pull request #854 from Altinity/fixes/antlya-25.3_version_fix

25.3 Antalya version fix

strtgbb commented:
I ran the regression version suite, it sees two issues.
- There is no `altinityantalya` suffix
- `(altinity build)` does not appear with the stack trace, is this due to an upstream change?

https://github.com/Altinity/clickhouse-regression/actions/runs/17218754598/job/48848782930
https://altinity-test-reports.s3.amazonaws.com/clickhouse/25.6.5.1/17218754598/testflows/x86_64/with_analyzer/keeper/without_thread_fuzzer/version/nice-new-fails.log.txt

Enmk committed:
Hackish attempt to bring version-related altinity modifications

From old CI code to the new CI code simpliy by reusing code from
'tests/ci/version_helper.py' and 'tests/ci/git_helper.py'

Enmk committed:
minor: fixed typos

Enmk committed:
Not using git_helper for information about number of commits.

Using almost original tweak updating code, some hacks to preserve
git_hash, get proper `description` and `string` values for version dict

Enmk committed:
Using proper version to build binaries

Enmk committed:
Attempt to fix the docker image tag to match binary version

Enmk committed:
another attempt

Enmk committed:
Another attempt

Enmk committed:
Fixed stupid error
